packages/react-dom-bindings/src/client/ReactFiberConfigDOM.js

```diff
@@ -1207,18 +1220,252 @@ export function cancelRootViewTransitionName(rootContainer: Container): void {
 }
 
 export function restoreRootViewTransitionName(rootContainer: Container): void {
+  let containerInstance: Instance;
+  if (rootContainer.nodeType === DOCUMENT_NODE) {
+    containerInstance = (rootContainer: any).body;
+  } else if (rootContainer.nodeName === 'HTML') {
+    containerInstance = (rootContainer.ownerDocument.body: any);
+  } else {
+    // If the container is not the whole document, then we ideally should probably
+    // clone the whole document outside of the React too.
```

Comment:
Whats the downside of only cloning the subtree affected? Seems like it would be more efficient. Is it that there could be parents affecting layout?

Reply:
Yea. If the root container itself changes size and that causes parents to relayout then really the parent tree should cross-fade. That's what happens with fire-and-forget View Transitions.